src/index.js

```diff
@@ -140,7 +140,7 @@
                             });
                         });
                 },
-                putUser (username, metadata = {}, db = databases[ defaultDB ]) {
+                putUser (username, db = databases[ defaultDB ], metadata = {}) {
```

Comment:
This is good because of consistency. Hopefully this won't break someones app.

Reply:
I moved the metadata parameter to the end for the TypeScript typings. TypeScript only allows optional parameters at the end and so the typing for the putUser method needed to be:

putUser (username: string, db: any, metadata?: {}): any;

The following generates a TS1016 error 'A required parameter cannot follow an optional parameter.'

putUser (username: string, metadata?: {}, db: any): any;

Ref: https://www.logicbig.com/tutorials/misc/typescript/function-optional-and-default-params.html

This was the only method I had this issue with.

I intend on improving the typings for all these methods in the future, this is just an initial effort. I'd like to re-use some of the typings already created for @types/pouchdb-core but currently I can't seem to access them since they're not exported. I'm looking into how to get that to work. But I think these typings will do for now.

Reply:
Actually, I think I can fix this by making all the db parameters optional as well, which they are. Then the vue.js plugin option defaultDB is required, or else you'd end up with no databases[defaultDB] to fall back on.

Reply:
I'm finding some inconsistencies with the parameter db in many of the methods. Sometimes db has to be of type string and then you can't have the default parameter db = databases[defaultDB] and instead will have the following code: 

if (!databases[db]) {
                        makeInstance(db);
                    }

other times the parameter db needs to be of type PouchDB and then you can have a default parameter of db = databases[defaultDB] and cannot have the following code because you aren't getting the right type for makeInstance: 

if (!databases[db]) {
                        makeInstance(db);
                    }

Let me do another commit.

If I typed makeInstance I could get all these type errors I think.